### PR TITLE
Updated openshift-cluster-autoscaler role and playbooks

### PR DESCRIPTION
**Pull request by frobware**
Various updates to the existing cluster-autoscaler role.

Additionally:
- hooked up by creating playbooks/openshift-cluster-autoscaler/...
- adds the cluster-autoscaler to components.yml
- enabled by setting `openshift_cluster_autoscaler_deploy: true` (default: `false`)
- only enabled for compute nodes

### CODE REVIEW

**Comment by vrutkovs** [comment]
Fails with `openshift_cluster_autoscaler_aws_key is undefined` on default install here, it should default to `openshift_cloudprovider_aws_access_key`

File: roles/openshift_cluster_autoscaler/templates/cluster-autoscaler.j2 [hunk 1]
```diff
@@ -20,8 +20,9 @@ spec:
     spec:
       containers:
       - args:
+        - /bin/cluster-autoscaler
         - --alsologtostderr
-        - --v={{ openshift_cluster_autoscaler_version }}
+        - --v={{ openshift_cluster_autoscaler_loglevel }}
         - --skip-nodes-with-local-storage={{ openshift_cluster_autoscaler_skip_nodes_local_storage }}
         - --leader-elect-resource-lock=configmaps
         - --namespace={{ openshift_cluster_autoscaler_namespace }}
```
Comment (ingvagabund) [inline roles/openshift_cluster_autoscaler/templates/cluster-autoscaler.j2]: So we still need a way to change the `--scale-down-delay-after-failure`, `--scale-down-unneeded-time` and `--scale-down-delay-after-add` before we deploy the autoscaler so we get faster response. What about this:
```yaml
{% if openshift_cluster_autoscaler_scale_down_delay_after_failure is defined %}
        - --scale-down-delay-after-failure={{ openshift_cluster_autoscaler_scale_down_delay_after_failure }}
{% endif %}
{% if openshift_cluster_autoscaler_scale_down_unneeded_timeout is defined %}
        - --scale-down-unneeded-time={{ openshift_cluster_autoscaler_scale_down_unneeded_timeout }}
{% endif %}
{% if openshift_cluster_autoscaler_scale_down_delay_after_add is defined %}
        - --scale-down-delay-after-add={{ openshift_cluster_autoscaler_scale_down_delay_after_add }}
{% endif %}
```

?

**Comment by ingvagabund** [comment]
@frobware can you create `examples` directory under the role dir and fetch `scale-up.yaml` file [1] there? So we can download the file to the master and run `oc create ...` over it.

[1] https://github.com/openshift/openshift-ansible/pull/9078/files#diff-37ac2ce7f4485779df60d12f620198ac

**Comment by ingvagabund** [comment]
> > Fails with openshift_cluster_autoscaler_aws_key is undefined on default install here, it should default to openshift_cloudprovider_aws_access_key (same for _secret)
> 
> Can you explain this a little more. Are you looking for a change?
Adding 
```yaml
- name: Check the AWS credentials are set
    fail:
      msg: "openshift_cluster_autoscaler_aws_key and/or openshift_cluster_autoscaler_aws_secret_key are not set"
    when: openshift_cluster_autoscaler_aws_secret_key is undefined or openshift_cluster_autoscaler_aws_secret_key == "" or openshift_cluster_autoscaler_aws_secret_key is undefined or openshift_cluster_autoscaler_aws_secret_key == ""
```

under tasks/aws.yml right before the `template` task will do the trick.

**Comment by vrutkovs** [comment]
>Can you explain this a little more. Are you looking for a change?

@frobware since it would be enable by default I'd expect it work with default settings, though its not related to this PR.

In default.yml we'd need to set default key and secret:
```
openshift_cluster_autoscaler_aws_key: "{{ openshift_cloudprovider_aws_access_key }}"
openshift_cluster_autoscaler_aws_secret: "{{ openshift_cloudprovider_aws_access_secret }}"
```
(these can be redefined later on).

**Comment by frobware** [comment]
> can you create examples directory under the role dir and fetch scale-up.yaml file [1] there? So we can download the file to the master and run oc create ... over it in the CI.

I think we concluded to do all this in the CI job. True?

File: roles/openshift_cluster_autoscaler/templates/cluster-autoscaler.j2 (hunk 1 above)
Comment (frobware) [reply roles/openshift_cluster_autoscaler/templates/cluster-autoscaler.j2]: > What about this:

Done.

**Comment by frobware** [comment]
> Fails with openshift_cluster_autoscaler_aws_key is undefined on default install here, it should default to openshift_cloudprovider_aws_access_key (same for _secret)

Add defaults and a sanity check to assert they are set. Please take another look.

**Comment by frobware** [comment]
@sdodson PTAL

File: roles/openshift_cluster_autoscaler/defaults/main.yml [hunk 2]
```diff
@@ -1,25 +1,27 @@
...
 openshift_cluster_autoscaler_serviceaccount: cluster-autoscaler
 openshift_cluster_autoscaler_region: us-east-1
 
-openshift_cluster_autoscaler_image: docker.io/openshift/kubernetes-autoscaler:v0.6.1
+openshift_cluster_autoscaler_image: "{{ l_os_registry_url | regex_replace('${component}' | regex_escape, 'cluster-autoscaler') }}"
 
 openshift_cluster_autoscaler_cloud_provider: aws
 
 # AWS specific variables here
+openshift_cluster_autoscaler_aws_key: "{{ openshift_cloudprovider_aws_access_key }}"
```
Comment (ingvagabund) [inline roles/openshift_cluster_autoscaler/defaults/main.yml]: openshift_cluster_autoscaler_aws_key: "{{ openshift_cloudprovider_aws_access_key | default('') }}"

File: roles/openshift_cluster_autoscaler/defaults/main.yml [hunk 3]
```diff
@@ -1,25 +1,27 @@
...
+openshift_cluster_autoscaler_aws_secret_key: "{{ openshift_cloudprovider_aws_secret_key }}"
```
Comment (ingvagabund) [inline roles/openshift_cluster_autoscaler/defaults/main.yml]: openshift_cluster_autoscaler_aws_secret_key: "{{ openshift_cloudprovider_aws_secret_key | default('') }}"

Comment (frobware) [reply roles/openshift_cluster_autoscaler/defaults/main.yml]: Done

**Comment by ingvagabund** [comment]
Verified in CI we can run the autoscaler role, deploy it and scale up and down. Let's merge this and squash it.

**Comment by vrutkovs** [comment]
Squashed the original commits

/lgtm

**Comment by derekwaynecarr** [comment]
woot!

**Pull request merged**